typing/printtyp.ml

```diff
@@ -884,6 +884,15 @@ let set_printing_env env =
     printing_cont := [cont];
   end
 
+(* CR-soon zqian: Currently we immediately backtrack each mutation, which might
+cause incoherent types/modes in a single printing. Instead, we should move
```

Comment:
This issue was already present before this PR, but if you do have a test on hand that demonstrates it, could you add it? If you don't, feel free to resolve this.

Reply:
Sadly I don't have an example on hand.